### PR TITLE
Wire logging via JSON Logger end-to-end.

### PR DESCRIPTION
**Pull request by ozevren**
+ Add a Name() method on AdapterConfig, so that appropriate adapter instance can be created by the adapterManager.
+ Add missing support in adapterManager for Logger registration and debugging.
+ Implement a basic logic for GetLoggerAdapter. This doesn't cache or take DispatchKey into account.
+ Add a basic, incomplete proto => internal log format.
+ Add a quick hack to GetLoggerAdapterConfigs to return the config for JSON Logger.
+ Add client support for making Report calls.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/78)

### CODE REVIEW

File: example/client/cmd/report.go [hunk 1]
```diff
@@ -15,17 +15,77 @@
...
+
+		var attrs map[string]string
+		if attrs, err = parseAttributes(Attributes); err != nil {
+			errorf(err.Error())
+			return
+		}
+
+		stream, err := cs.client.Report(context.Background())
+		if err != nil {
+			errorf("Check RPC failed: %v", err)
```
Comment (geeknoid) [inline example/client/cmd/report.go]: Check -> Report

File: example/client/cmd/report.go [hunk 2]
```diff
@@ -15,17 +15,77 @@
...
-	},
-}
+		if len(args) == 0 {
+			errorf("Message is missing.")
+			return
+		}
+
+		cs, err := createAPIClient(MixerAddress)
+		if err != nil {
+			errorf("Unable to establish connection to %s", MixerAddress)
```
Comment (geeknoid) [inline example/client/cmd/report.go]: Include 'err' in message

File: example/client/cmd/report.go [hunk 3]
```diff
@@ -15,17 +15,77 @@
...
+		}
+
+		if err := stream.Send(&request); err != nil {
+			errorf("Failed to send Report RPC: %v", err)
+			return
+		}
+
+		response, err := stream.Recv()
+		if err == io.EOF {
+			errorf("Got no response from Check RPC")
```
Comment (geeknoid) [inline example/client/cmd/report.go]: Check->Report

**Comment by ozevren** [comment]
Updated. PTAL.

File: example/client/cmd/report.go [hunk 4]
```diff
@@ -15,17 +15,77 @@
...
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
 	"github.com/spf13/cobra"
+	mixerpb "istio.io/mixer/api/v1"
```
Comment (douglas-reid) [inline example/client/cmd/report.go]: please separate out the named imports from the unnamed imports.

File: example/client/cmd/report.go [hunk 5]
```diff
@@ -15,17 +15,77 @@
 package cmd
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
```
Comment (douglas-reid) [inline example/client/cmd/report.go]: I think you need to add this dependency to the BUILD file (based on the Travis errors).

File: server/apiHandlers.go [hunk 6]
```diff
@@ -18,6 +18,7 @@ package main
 
 import (
 	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes"
```
Comment (douglas-reid) [inline server/apiHandlers.go]: As above, I think you need to add the appropriate lib to the BUILD file (based on Travis errors).

File: example/client/cmd/report.go (hunk 5 above)
Comment (ozevren) [inline example/client/cmd/report.go]: Thanks for the review. My local bazel installation is not working. Do you know what this error means:



ozben-macbookpro:mixer ozben$ bazel build ...

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:500:18: expected ConfigurationTransition or NoneType for 'cfg' while calling label but got string instead: host.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:528:20: name 'go_env_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:560:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 558

rule(go_library_impl, attrs = go_librar..."])}, <2 more arguments>)

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 560, in rule

go_library_attrs

name 'go_library_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:571:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 569

rule(go_binary_impl, attrs = go_library...()}, <3 more arguments>)

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 571, in rule

go_library_attrs

name 'go_library_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:582:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 580

rule(go_test_impl, attrs = go_library_a...()}, <4 more arguments>)

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 582, in rule

go_library_attrs

name 'go_library_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:686:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 684

rule(_cgo_codegen_impl, attrs = go_env_...)}, <2 more arguments>)

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 686, in rule

go_env_attrs

name 'go_env_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:798:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 796

rule(_cgo_import_impl, attrs = go_env_a..."), <2 more arguments>)})

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 798, in rule

go_env_attrs

name 'go_env_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:1044:20: expected ConfigurationTransition or NoneType for 'cfg' while calling label but got string instead: host.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:1055:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 1053

repository_rule(implementation = _go_repository_..., ...)

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 1055, in repository_rule

_go_repository_attrs

name '_go_repository_attrs' is not defined.

ERROR: /private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl:1060:13: Traceback (most recent call last):

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 1058

repository_rule(implementation = _new_go_reposit..., ..."), <4 more arguments>)})

File "/private/var/tmp/_bazel_ozben/8f37a417b093a76dc66e776e0ddfb7f1/external/io_bazel_rules_go/go/def.bzl", line 1060, in repository_rule

_go_repository_attrs

name '_go_repository_attrs' is not defined.

ERROR: com.google.devtools.build.lib.packages.BuildFileContainsErrorsException: error loading package '': Extension file 'go/def.bzl' has errors.

INFO: Elapsed time: 0.127s

ozben-macbookpro:mixer ozben$

Comment (douglas-reid) [inline example/client/cmd/report.go]: What version of bazel are you running locally?  Anything after 0.3.1 should be OK, as far as I can tell.

**Pull request merged**